monitor_ingest/actions/setup_entity.py

```diff
@@ -0,0 +1,99 @@
+import yaml
+from st2common.runners.base_action import Action
+from mam.sdk import (entitytype, constants, kpifunction, dimension)
+
+__all__ = ['SetupEntityAction']
+
+
+class SetupEntityAction(Action):
+    def __init__(self, config):
+        super(SetupEntityAction, self).__init__(config)
+        self._config = self.config
+        self._credentials = self._config.get('credentials', None)
+        self._data_file_path = self._config.get('data_file_path', None)
+        self._action_type = self._config.get('action_type', None)
+
+        if not self._config:
+            raise ValueError('Missing config yaml')
+        if not self._credentials:
+            raise ValueError('Missing IBM Watson IoT credentials in config file')
+        if not self._data_file_path:
+            raise ValueError('Missing data file path in config file')
+        if not self._action_type:
+            raise ValueError('Missing action type key in config file')
+
+    def run(self):
+        operations_completed = {}
```

Comment:
I do not like using a mutable dictionary as a global variable. That's not intuitive and difficult to debug. Since all of the `setup_*` functions are mutually exclusive, simply have them return the result you would like to return to StackStorm.

Reply:
removed mutable dictionary